planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/clothoid_pull_out.cpp

```diff
@@ -0,0 +1,1680 @@
+    A_min, L_min, initial_velocity);
+
+  // Execute original clothoid conversion
+  auto clothoid_points_opt =
+    convert_arc_to_clothoid(arc_segment, start_pose, A_min, L_min, point_interval);
+
+  if (!clothoid_points_opt) {
+    RCLCPP_DEBUG(
+      rclcpp::get_logger("ClothoidPullOut"),
+      "Clothoid conversion failed! Check parameters and arc segment validity.");
+    RCLCPP_DEBUG(
+      rclcpp::get_logger("ClothoidPullOut"),
+      "Arc segment: radius=%.3f, center=(%.3f, %.3f), is_clockwise=%s", arc_segment.radius,
+      arc_segment.center.x, arc_segment.center.y, arc_segment.is_clockwise ? "true" : "false");
+    return std::nullopt;
+  }
+
+  // Apply endpoint correction
+  auto corrected_points =
+    correct_clothoid_by_rigid_transform(*clothoid_points_opt, arc_segment, start_pose);
+
+  return corrected_points;
+}
+
+// std::optional<PathWithLaneId> might be better
+PathWithLaneId create_path_with_lane_id_from_clothoid_paths(
+  const std::vector<std::vector<geometry_msgs::msg::Point>> & clothoid_paths,
+  const geometry_msgs::msg::Pose & target_pose, double velocity, double target_velocity,
+  double acceleration, const lanelet::ConstLanelets & road_lanes,
+  const std::shared_ptr<autoware::route_handler::RouteHandler> & route_handler)
+{
+  (void)target_pose;  // Suppress unused parameter warning
+  // Return empty PathWithLaneId if clothoid paths are empty
+  if (clothoid_paths.empty()) {
+    PathWithLaneId empty_path;
+    empty_path.header = route_handler->getRouteHeader();
+    return empty_path;
+  }
+
+  // Combine all clothoid paths
+  std::vector<geometry_msgs::msg::Point> all_clothoid_points;
+  for (const auto & path : clothoid_paths) {
+    // Process only if path is not empty
+    if (path.empty()) {
+      continue;
+    }
+
+    // Skip first point for paths other than the first (avoid duplication)
+    // Is it okay to skip?
+    size_t start_idx = (all_clothoid_points.empty()) ? 0 : 1;
+    for (size_t j = start_idx; j < path.size(); ++j) {
+      all_clothoid_points.push_back(path[j]);
+    }
+  }
+
+  // Return empty PathWithLaneId if still empty after combination
+  if (all_clothoid_points.empty()) {
+    PathWithLaneId empty_path;
+    empty_path.header = route_handler->getRouteHeader();
+    return empty_path;
+  }
+
+  // Create PathWithLaneId
+  PathWithLaneId path_with_lane_id;
+  path_with_lane_id.header = route_handler->getRouteHeader();
+
+  // Convert each coordinate point to PathPointWithLaneId
+  for (size_t i = 0; i < all_clothoid_points.size(); ++i) {
+    PathPointWithLaneId path_point;
+
+    // Set coordinates
+    path_point.point.pose.position = all_clothoid_points[i];
+
+    // Set z coordinate: get z value of closest point from lanelet information
+    if (!road_lanes.empty()) {
+      // Find closest lanelet
+      lanelet::Lanelet closest_lanelet;
+      if (lanelet::utils::query::getClosestLanelet(
+            road_lanes, path_point.point.pose, &closest_lanelet)) {
```

Comment:
Even if height of road_shoulder and road is different, road_shoulder is not used for `z` ?

Reply:
The arguments were confusing, so I changed them in https://github.com/autowarefoundation/autoware_universe/pull/11040/commits/74e5e8c06654b76de0e880c8ee78c7a808323122
Since `search_lanes` includes road_shoulders, it's expected that the z-coordinates of the road_shoulder are also taken into consideration.